src/java.base/share/classes/jdk/internal/classfile/impl/BlockCodeBuilderImpl.java

```diff
@@ -72,11 +72,10 @@ public boolean isEmpty() {
     }
 
     private int topLocal(CodeBuilder parent) {
-        return switch (parent) {
-            case BlockCodeBuilderImpl b -> b.topLocal;
-            case ChainedCodeBuilder b -> b.terminal.curTopLocal();
-            case TerminalCodeBuilder b -> b.curTopLocal();
-        };
+        if (parent instanceof BlockCodeBuilderImpl bcb) {
+            return bcb.topLocal;
+        }
+        return findTerminal(parent).curTopLocal();
```

Comment:
This advanced switch prevents `CodeBuilder::ifThen` or `ifThenElse` from being used in early bootstrap.